docs/notebooks/analysis/fit_functions.ipynb

```diff
@@ -291,7 +291,7 @@
     "plt.legend(fontsize=14, loc=\"upper left\")\n",
     "\n",
     "txt = f\"$f(x) = {explin.latex_str}$\\n$r^2 = {explin.rsq:.3f}$\\n\"\n",
-    "for name, param, err in zip(explin.param_names, explin.params, explin.param_errors):\n",
+    "for name, param, err in zip(explin.param_names, explin.params, explin.param_errors, strict=False):\n",
```

Comment:
The `strict=False` flag for `zip` allows the zipped iterables to have different lengths. In general, it will probably be better to set `strict=True` for most of the time that we use `zip`.  However, I'm not doing that here, since it will take a bit of work to think about each usage of `zip` to make sure it is desirable to set `strict=True`.  I plan to raise an issue about `strict=True` after this is merged.

Reply:
Would it be easier to set strict to False as the default in Zip for the time being then? Else wont all these strict=False statements have to be altered later down the line?

Reply:
This came from ruff rule [`B905`](https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/) which requires an explicit `strict=True` or `strict=False` in calls to `zip`. I decided against changing current behavior in this PR, and keeping `strict=False` will make the code more explicit in what it's actually doing.  You're right though that most (all?) of the `strict=False` statements will need to be altered later on.